openpype/hosts/substancepainter/plugins/create/create_textures.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+"""Creator plugin for creating textures."""
+
+from openpype.pipeline import CreatedInstance, Creator
+from openpype.lib import (
+    EnumDef,
+    UILabelDef,
+    NumberDef
+)
+
+from openpype.hosts.substancepainter.api.pipeline import (
+    set_project_metadata,
+    get_project_metadata
+)
+from openpype.hosts.substancepainter.api.lib import get_export_presets
+
+import substance_painter.project
+
+
+class CreateTextures(Creator):
+    """Create a texture set."""
+    identifier = "io.openpype.creators.substancepainter.textureset"
+    label = "Textures"
+    family = "textureSet"
+    icon = "picture-o"
+
+    default_variant = "Main"
+
+    def create(self, subset_name, instance_data, pre_create_data):
+
+        if not substance_painter.project.is_open():
+            return
+
+        instance = self.create_instance_in_context(subset_name, instance_data)
+        set_project_metadata("textureSet", instance.data_to_store())
+
+    def collect_instances(self):
+        workfile = get_project_metadata("textureSet")
+        if workfile:
+            self.create_instance_in_context_from_existing(workfile)
+
+    def update_instances(self, update_list):
+        for instance, _changes in update_list:
+            # Update project's metadata
+            data = get_project_metadata("textureSet") or {}
+            data.update(instance.data_to_store())
+            set_project_metadata("textureSet", data)
+
+    def remove_instances(self, instances):
+        for instance in instances:
+            # TODO: Implement removal
+            # api.remove_instance(instance)
+            self._remove_instance_from_context(instance)
+
+    # Helper methods (this might get moved into Creator class)
+    def create_instance_in_context(self, subset_name, data):
+        instance = CreatedInstance(
+            self.family, subset_name, data, self
+        )
+        self.create_context.creator_adds_instance(instance)
+        return instance
+
+    def create_instance_in_context_from_existing(self, data):
+        instance = CreatedInstance.from_existing(data, self)
+        self.create_context.creator_adds_instance(instance)
+        return instance
+
+    def get_instance_attr_defs(self):
+
+        return [
+            EnumDef("exportPresetUrl",
+                    items=get_export_presets(),
+                    label="Output Template"),
+            EnumDef("exportFileFormat",
+                    items={
+                        None: "Based on output template",
+                        # TODO: implement extensions
```

Comment:
Could this be defined using settings? If substance have limitations, than maybe multiselection enum?

Reply:
It could technically be implemented in settings. We could also always enforce it to be based on output template (which it is now due to not having the extensions enum) - which means the admin can just make sure the output templates are as required for production. 👍